server/src/main/java/org/opensearch/index/query/AbstractQueryBuilder.java

```diff
@@ -125,6 +134,7 @@ public final Query toQuery(QueryShardContext context) throws IOException {
                 context.addNamedQuery(queryName, query);
             }
         }
+        fieldType = context.fieldMapper(fieldName()).typeName();
```

Comment:
What does the fieldMapper return for none field? Maybe have check to ensure it is non-default value?

Reply:
I checked the underlying function calls and updated to return `null` if the field has no mapping.

*I will revisit this edge case once we finalize the default behavior of `fieldName()`